include/s3select.h

```diff
@@ -505,12 +530,22 @@ struct s3select : public bsc::grammar<s3select>
 
       arithmetic_argument = (float_number)[BOOST_BIND_ACTION(push_float_number)] |  (number)[BOOST_BIND_ACTION(push_number)] | (column_pos)[BOOST_BIND_ACTION(push_column_pos)] |
                             (string)[BOOST_BIND_ACTION(push_string)] |
-                            (cast) |
+                            (cast) | (trim) |
                             (function) | (variable)[BOOST_BIND_ACTION(push_variable)] ;//function is pushed by right-term
 
       cast = (bsc::str_p("cast") >> '(' >> arithmetic_expression >> bsc::str_p("as") >> (data_type)[BOOST_BIND_ACTION(push_data_type)] >> ')') [BOOST_BIND_ACTION(push_cast_expr)];
 
       data_type = (bsc::str_p("int") | bsc::str_p("float") | bsc::str_p("string") |  bsc::str_p("timestamp") );
+      
+      trim = (trim_one) | (trim_two) | (trim_three);
+
+      trim_two = (bsc::str_p("trim") >> '(' >> (trim_type)[BOOST_BIND_ACTION(push_trim_type)] >> arithmetic_expression >> ')') [BOOST_BIND_ACTION(push_trim_expr_two)];
+
+      trim_one = (bsc::str_p("trim") >> '(' >> arithmetic_expression >> ')') [BOOST_BIND_ACTION(push_trim_expr_one)];
+
+      trim_three = (bsc::str_p("trim") >> '(' >> ((trim_type)[BOOST_BIND_ACTION(push_trim_type)] >> arithmetic_expression >> arithmetic_expression)  >> ')') [BOOST_BIND_ACTION(push_trim_expr_three)];
+      
+      trim_type = (bsc::str_p("leading from") | bsc::str_p("trailing from") | bsc::str_p("both from") | bsc::str_p("from") );
```

Comment:
as commented, the /from/ syntax rule, is not handle correctly.
should verify whether its mandatory, and accordingly to place a correct rule.
(optional or mandatory)